### PR TITLE
VZ-4643 - Add option to configure replica count for AuthProxy deployment

### PR DESCRIPTION
**Pull request by mgianatagh**
# Description

Add option to configure replica count for AuthProxy deployment.  

Set the default replica count to 2 for prod deployments.

Fixes VZ-4643

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go [hunk 1]
```diff
@@ -378,6 +378,12 @@ type OAMComponent struct {
 type VerrazzanoComponent struct {
 	// +optional
 	Enabled *bool `json:"enabled,omitempty"`
+
+	// Arguments for installing Verrazzano
+	// +optional
+	// +patchMergeKey=name
+	// +patchStrategy=merge,retainKeys
+	VerrrazzanoInstallArgs []InstallArgs `json:"installArgs,omitempty" patchStrategy:"merge,retainKeys" patchMergeKey:"name"`
```
Comment (pfmackin) [inline platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go:386]: can this just be InstallArgs?  It already has context being in the VerrazzanoComponent

**Review by pfmackin — APPROVED** [review]

File: platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy.yaml [hunk 2]
```diff
@@ -30,7 +30,7 @@ metadata:
   name: {{ .Values.api.name }}
   namespace: {{ .Release.Namespace }}
 spec:
-  replicas: 1
+  replicas: {{ .Values.api.replicas }}
```
Comment (michael-cico) [inline platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy.yaml:33]: nit, should this be `authproxy.replicas`?

Comment (mgianatagh) [reply platform-operator/helm_config/charts/verrazzano/templates/verrazzano-authproxy.yaml:33]: The values.yaml already has an "api" section with things like "api.name" set to verrazzano-authproxy.

File: platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go (hunk 1 above)
Comment (mgianatagh) [reply platform-operator/apis/verrazzano/v1alpha1/verrazzano_types.go]: fixed

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**